roles/influxdb/tasks/bucket.yml

```
@@ -51,19 +51,33 @@
 - name: Create token für bucket
   shell: >
     influx auth create
+    --read-bucket "{{ bucket_id.stdout }}"
     --write-bucket "{{ bucket_id.stdout }}"
     |grep -v ^ID
     | awk '{print $2}'
   register: bucket_token
   when: token_list.rc != 0
 
+- name: Bucket mapping
+  shell: >
+    influx v1 dbrp create
+    --db "{{ influxdb_bucket }}"
+    --rp "{{ influxdb_bucket }}"
+    --bucket-id "{{ bucket_id.stdout}}"
+  when: (influxdb_influxql_query) and (result.rc != 0)
+
 - name: Read tokens
   shell: >
     influx auth list
     | grep "{{ bucket_id.stdout}}"
     | awk '{print $2}'
   register: token_list
   changed_when: false
-- name: Save token
-  set_fact:
-    token: "{{ token_list.stdout }}"
+
+- name: Save token in file
+  copy:
+    content: "{{ token_list.stdout }}"
+    dest: /root/token
+    owner: root
+    group: root
+    mode: '0644'
```

```suggestion
    dest: "/root/influxdb_token_{{ bucket_id.stdout }}"
    owner: root
    group: root
    mode: '0600'
```
Maybe use a more general path for the token. Honestly, I don't know how a bucket id looks like and if that makes any sense at all. All I wanted to show is that it would beneficial to:
1. have in the filename what the token is used for. Think of hosts that run several services together
2. have the option to use several token for different databases/buckets/whatever

First I even considered replacing `/root/` with `/opt/`. But that doesn't feel right, either. Maybe use a variable which defaults to `/root/`. This way users could make their own path if they want to.